src/File.php

```diff
@@ -4,467 +4,340 @@
+            compressionMethod: $this->compressionMethod,
+            lastModificationDateTime: $this->lastModificationDateTime,
+            crc32UncompressedData: $this->crc,
+            compressedSize: ($this->enableZip64 || $this->enableZeroHeader || $this->compressedSize > 0xFFFFFFFF)
+                ? 0xFFFFFFFF
+                : $this->compressedSize,
+            uncompressedSize: ($this->enableZip64 || $this->enableZeroHeader || $this->uncompressedSize > 0xFFFFFFFF)
+                ? 0xFFFFFFFF
+                : $this->uncompressedSize,
+            fileName: $this->fileName,
+            extraField: $footer,
+        );
 
-        $footer = $this->buildZip64ExtraBlock($force);
 
-        // If this file will start over 4GB limit in ZIP file,
-        // CDR record will have to use Zip64 extension to describe offset
-        // to keep consistency we use the same value here
-        if ($this->zip->ofs->isOver32()) {
-            $this->version = Version::ZIP64();
-        }
+        ($this->send)($data);
 
-        $fields = [
-            ['V', ZipStream::FILE_HEADER_SIGNATURE],
-            ['v', $this->version->getValue()],      // Version needed to Extract
-            ['v', $this->bits],                     // General purpose bit flags - data descriptor flag set
-            ['v', $this->method->getValue()],       // Compression method
-            ['V', $time],                           // Timestamp (DOS Format)
-            ['V', $this->crc],                      // CRC32 of data (0 -> moved to data descriptor footer)
-            ['V', $this->zlen->getLowFF($force)],   // Length of compressed data (forced to 0xFFFFFFFF for zero header)
-            ['V', $this->len->getLowFF($force)],    // Length of original data (forced to 0xFFFFFFFF for zero header)
-            ['v', $nameLength],                     // Length of filename
-            ['v', strlen($footer)],                 // Extra data (see above)
-        ];
-
-        // pack fields and calculate "total" length
-        $header = ZipStream::packFields($fields);
-
-        // print header and filename
-        $data = $header . $name . $footer;
-        $this->zip->send($data);
-
-        // save header length
-        $this->hlen = Bigint::init(strlen($data));
+        $this->totalSize +=  strlen($data);
     }
 
     /**
      * Strip characters that are not legal in Windows filenames
      * to prevent compatibility issues
-     *
-     * @param string $filename Unprocessed filename
-     * @return string
      */
-    public static function filterFilename(string $filename): string
-    {
+    public static function filterFilename(
+        /**
+         * Unprocessed filename
+         */
+        string $fileName
+    ): string {
         // strip leading slashes from file name
         // (fixes bug in windows archive viewer)
-        $filename = preg_replace('/^\\/+/', '', $filename);
+        $fileName = preg_replace('/^\\/+/', '', $fileName);
```

Comment:
I was going to suggest `'%^/+%'` but I think this would be even cleaner/quicker

```suggestion
        $fileName = ltrim($fileName, '/');
```